libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -399,6 +402,42 @@ void CodeTransform::operator()(Switch const& _switch)
 	m_assembly.appendInstruction(evmasm::Instruction::POP);
 }
 
+
+namespace
+{
+
+/// Shuffles @a _container by swapping and popping its elements using the
+/// @a _swap and @a _pop functions in order to remove the elements for which
+/// @a _keep returns false.
+template<typename C, typename KeepFn, typename SwapFn, typename PopFn>
+void shuffle(C& _container, KeepFn&& _keep, SwapFn&& _swap, PopFn&& _pop)
+{
+	set<int> slotsToBeRemoved;
+	for (auto&& [n, elem]: ranges::views::enumerate(_container))
+		if (!_keep(elem))
+			slotsToBeRemoved.emplace(static_cast<int>(n));
+
+	while (!slotsToBeRemoved.empty())
+	{
+		if (_keep(_container.back()))
+		{
+			yulAssert(!slotsToBeRemoved.empty(), "");
+			int slot = *slotsToBeRemoved.begin();
+			slotsToBeRemoved.erase(slotsToBeRemoved.begin());
+
+			_swap(slot);
```

Comment:
It might be more readable if `container` gets passed as an argument. Or if the actual swaps are done at the end, somehow.

Reply:
Passing it as argument would be fine. Doing it later is not possible.

Reply:
So now the shuffling is independent of the concrete container in general. Is that better?